Packs/PrismaSaasSecurity/Integrations/SaasSecurityEventCollector/SaasSecurityEventCollector.py

```diff
@@ -0,0 +1,254 @@
+
+    Returns:
+        bool: True if token has expired, False if not.
+    """
+    return time.time() - token_initiate_time >= token_expiration_seconds - 60
+
+
+def validate_limit(limit: int):
+    """
+    Validate that the limit/max fetch is a number divisible by the MAX_EVENTS_PER_REQUEST (100).
+    """
+    if limit % MAX_EVENTS_PER_REQUEST != 0:
+        raise DemistoException(f'fetch limit parameter should be divisible by {MAX_EVENTS_PER_REQUEST}')
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client):
+    """
+    Testing we have a valid connection to Saas-Security.
+
+    The reason not testing fetching events is that if we fetch it from the api we will lose those requests.
+    """
+    response = client.get_event_request()
+    if response.status_code in (200, 204):
+        return 'ok'
+
+
+def saas_security_get_events_command(client: Client, args: Dict) -> Union[str, CommandResults]:
+    """
+    Fetches events from the saas-security queue and return them to the war-room.
+    in case should_push_events is set to True, they will be also sent to XSIAM.
+    """
+    limit = arg_to_number(args.get('limit')) or 100
+    validate_limit(limit=limit)
+    should_push_events = argToBoolean(args.get('should_push_events'))
+
+    if events := fetch_events(client=client, max_fetch=limit):
+        if should_push_events:
+            send_events_to_xsiam(events=events, vendor='PaloAltoNetworks', product='SaasSecurity')
+        return CommandResults(
+            readable_output=tableToMarkdown(
+                'SaaS Security Logs',
+                events,
+                headers=['log_type', 'item_type', 'item_name', 'timestamp', 'serial'],
+                headerTransform=underscoreToCamelCase,
+                removeNull=True
+            ),
+            raw_response=events,
+            outputs=events,
+            outputs_key_field=['timestamp', 'log_type, item_name, item_type'],
+            outputs_prefix='SaasSecurity.Event'
+        )
+    return 'No events were found.'
+
+
+def fetch_events(client: Client, max_fetch: int) -> List[Dict]:
+    """
+    Fetches events from the saas-security queue.
+    """
+    events: List[Dict] = []
+    response = client.get_events_request()
+
+    while len(events) < max_fetch and response.status_code == 200:
+        fetched_events = response.json().get('events') or []
+        demisto.debug(f'fetched events: ({fetched_events})')
+        events.extend(fetched_events)
+        response = client.get_events_request()
+
+    return events
+
+
+def main() -> None:
+    params = demisto.params()
+    client_id: str = params.get('credentials', {}).get('identifier', '')
+    client_secret: str = params.get('credentials', {}).get('password', '')
+    base_url: str = params.get('url', '').rstrip('/')
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    max_fetch = arg_to_number(params.get('max_fetch')) or 100
```

Comment:
arg or param or 100

Reply:
args should be first in that list, so it takes precedence

`max_fetch = arg_to_number(args.get('limit') or params.get('max_fetch') ) or 100`

Reply:
done